src/bridge/views/AcceptTransfer.js

```diff
@@ -32,144 +32,17 @@ class AcceptTransfer extends React.Component {
 
     this.state = {
       receivingAddress: receivingAddress,
-      txn: Nothing(),
-      txError: Nothing(),
       incomingPoint: incomingPoint,
-      userApproval: false,
-      nonce: '',
-      gasPrice: '5',
-      showGasDetails: false,
-      chainId: '',
-      gasLimit: '600000',
-      stx: Nothing(),
     }
 
     this.handleAddressInput = this.handleAddressInput.bind(this)
-    this.handleCreateUnsignedTxn = this.handleCreateUnsignedTxn.bind(this)
-    this.handleSubmit = this.handleSubmit.bind(this)
-    this.handleSetUserApproval = this.handleSetUserApproval.bind(this)
-    this.handleSetTxn = this.handleSetTxn.bind(this)
-    this.handleSetStx = this.handleSetStx.bind(this)
-    this.handleSetNonce = this.handleSetNonce.bind(this)
-    this.handleSetChainId = this.handleSetChainId.bind(this)
-    this.handleSetGasPrice = this.handleSetGasPrice.bind(this)
-    this.handleSetGasLimit = this.handleSetGasLimit.bind(this)
-    this.toggleGasDetails = this.toggleGasDetails.bind(this)
-  }
-
-  toggleGasDetails() {
-    this.setState({
-      showGasDetails: !this.state.showGasDetails
-    })
-  }
-
-  componentDidMount() {
-    const { props } = this
-
-    const addr = props.wallet.matchWith({
-      Just: wal => addressFromSecp256k1Public(wal.value.publicKey),
-      Nothing: () => {
-        throw BRIDGE_ERROR.MISSING_WALLET
-      }
-    })
-
-    props.web3.matchWith({
-      Nothing: () => {},
-      Just: (w3) => {
-        const validWeb3 = w3.value
-
-        const getTxMetadata = [
-          validWeb3.eth.getTransactionCount(addr),
-          validWeb3.eth.net.getId(),
-          validWeb3.eth.getGasPrice()
-        ];
-
-        Promise.all(getTxMetadata).then(r => {
-          const txMetadata = {
-            nonce: r[0],
-            chainId: r[1],
-            gasPrice: fromWei(r[2], 'gwei'),
-          };
-
-          this.setState({...txMetadata})
-
-        })
-      }
-    });
+    this.createUnsignedTxn = this.createUnsignedTxn.bind(this)
+    this.statelessRef = React.createRef();
```

Comment:
The React docs say you need to add a `ref` attribute to some DOM element for this to work? I know this ends up with having a pointer to the `StatelessTransaction` in `statelessRef.current`, but how does that hook up? I don't see it happen anywhere.

Reply:
Ah, good catch! I missed this one. It's there now.